### PR TITLE
[bugs] fixes for lgtm and improve docs

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

* project Euler problem 22 lgtm error fix
* fix inconsistent use of `double` and `long double` in `durand_kerner_roots.c` 
* add docs & fix lgtm alerts in `bead_sort.c`
#### References

* #550 
* [project 22 - 1](https://lgtm.com/projects/g/TheAlgorithms/C/snapshot/4d0dbc401d67171d555d494c49a08e72303b5ba7/files/project_euler/problem_22/sol1.c?sort=name&dir=ASC&mode=heatmap#xa0d183509f9bbb35:1)
* [durange_kerner_roots](https://lgtm.com/projects/g/TheAlgorithms/C/snapshot/4d0dbc401d67171d555d494c49a08e72303b5ba7/files/numerical_methods/durand_kerner_roots.c?sort=name&dir=ASC&mode=heatmap)
* [bead_sort.c](https://lgtm.com/projects/g/TheAlgorithms/C/snapshot/4d0dbc401d67171d555d494c49a08e72303b5ba7/files/sorting/bead_sort.c?sort=name&dir=ASC&mode=heatmap#xdc9736f163b5f58d:1)

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by kvedala** [comment]
> This works too😂 I didn't see it this way.
> 
> There is another file `numerical_methods/durand_kerner_roots.c` which has a similar issue on line 50 where the type of `coeff` is `double` instead of `long double`.

rule of thumb: avoid typecasting and find and address the root-problem :)

**Comment by tjgurwara99** [comment]
I can approve this if you want to create a separate PR for `durand_kerner_roots.c` @kvedala

File: numerical_methods/durand_kerner_roots.c [hunk 1]
```diff
@@ -108,8 +108,8 @@ int main(int argc, char **argv)
     }
 
     degree = argc - 1; /* detected polynomial degree */
-    coeffs = (double *)malloc(
-        degree * sizeof(double)); /* store all input coefficients */
+    coeffs = (long double *)malloc(
+        degree * sizeof(long double)); /* store all input coefficients */
```
Comment (tjgurwara99) [inline numerical_methods/durand_kerner_roots.c:112]: printf might complain now

Comment (tjgurwara99) [reply numerical_methods/durand_kerner_roots.c:112]: oh LGTM issued the alert

Comment (kvedala) [reply numerical_methods/durand_kerner_roots.c:112]: yes, converted to draft. dont review until marked as ready for review.

**Pull request merged**